File: backend/src/main/java/com/bidbridge/backend/entity/Quote.java

package com.bidbridge.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;
import java.time.LocalDate;
import java.time.Instant;

@Data
@NoArgsConstructor
@Entity
@Table(name = "quotes", schema = "public")
public class Quote {

    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "job_id")
    private Job job;

    @ManyToOne
    @JoinColumn(name = "offering_company_id")
    private Company offeringCompany;

    private Double price;
    private String currency;
    private Integer transitTime;
    private LocalDate validUntil;

    private String note;
    private String address;
    private String status;

    private Instant createdAt;
}

File: frontend/src/types/quote.ts
export interface Quote {
  id: string;
  job_id: string;
  offering_company_id: string;
  price: number;
  currency: string;
  transit_time: number;
  valid_until: string;
  note: string;
  address: string;
  status: string;
  created_at: string;
}

export interface QuoteApiResponse {
  id: string;
  jobId: string;
  offeringCompanyId: string;
  price: number;
  currency: string;
  transitTime: number;
  validUntil: string;
  note: string;
  address: string;
  status: string;
  createdAt: string;
}

export function transformQuoteFromApi(apiQuote: QuoteApiResponse): Quote {
  return {
    id: apiQuote.id,
    job_id: apiQuote.jobId,
    offering_company_id: apiQuote.offeringCompanyId,
    price: apiQuote.price,
    currency: apiQuote.currency,
    transit_time: apiQuote.transitTime,
    valid_until: apiQuote.validUntil,
    note: apiQuote.note,
    address: apiQuote.address,
    status: apiQuote.status,
    created_at: apiQuote.createdAt,
  };
}
